src/main/java/org/myrobotlab/service/data/AbstractData.java

```diff
@@ -0,0 +1,17 @@
+package org.myrobotlab.service.data;
+
+/**
+ * An attempt to try to normalize some of the common fields of these pojos
+ * 
+ * @author grog
+ *
+ */
+public abstract class AbstractData {
+
+  public String source;
```

Comment:
It's a bit overkill to create an abstract base class for this...  don't you think?

Reply:
I don't think so - as soon as anyone gets more than one of the same type of sensor - regardless of the type, they will want to know the origin of the sensor data, and source/name can sort it out